### DISCUSSION TITLE
Working with Chaos Mesh in a namespace scope environment(Webhooks not working)

### DISCUSSION THREAD

**Discussion by sylosis91** [chaos-mesh/chaos-mesh#3908] [Q&A]
**What version of Kubernetes are you using?**
minikube version: v1.26.0
Kubernetes version: v1.24.1

**What version of Chaos Mesh are you using?**
2.5.0

**What did you do? / Minimal Reproducible Example**
Changing some chart values, specifically on Values.yaml:

clusterScoped: false
targetNamespace: xatops

**What did you expect to see?**
Experiments working the same way they did when clusterRole was true.

**What did you see instead?**
An error about permissions thrown by the webhooks. ontroller-runtime.webhook.webhooks     admission/http.go:97    received request        {"webhook": "/mutate-chaos-mesh-org-v1alpha1-podchaos", "UID": "91e29c36-ca7c-4199-bf15-e06c2a606b69", "kind": "[chaos-mesh.org/v1alpha1](http://chaos-mesh.org/v1alpha1), Kind=PodChaos", "resource": {"group":"[chaos-mesh.org](http://chaos-mesh.org/)","version":"v1alpha1","resource":"podchaos"}}

2023-01-26T17:16:28.658Z        DEBUG   controller-runtime.webhook.webhooks     admission/http.go:136   wrote response  {"webhook": "/mutate-chaos-mesh-org-v1alpha1-podchaos", "code": 200, "reason": "", "UID": "91e29c36-ca7c-4199-bf15-e06c2a606b69", "allowed": true}

We have a very restricted environment and we are trying to make it work within namespaces. I've seen that there are two objects (ClusterRole and a ClusterRoleBinding in Chaos-controller-manager-rbac.yaml) which both of them are always created even though clusteRole is false, they seem to be at cluster-wide level. Is there any way to make it work without having those 2 objects?

Thanks in advance.

**Comment by jpmargo** [comment]
Hi @cwen0, I see that you implemented this part in the commit https://github.com/chaos-mesh/chaos-mesh/commit/ea12c187dcc8ea7d2b26f46cb0cf88ad97b5faaf, could you help us?
Many thanks!
